### ISSUE TITLE
History.java public c'tor should either be 'more private' or documented to not be used by users.

### ISSUE THREAD

**Issue by enver-haase** [vaadin/flow#18511]
### Description of the bug

I can instantiate History as a Vaadin Flow user.

### Expected behavior

The History c'tor should probably be package-private or such or at least documented to not be used by the user.
Instead:
'''UI.getCurrent().getPage().getHistory()''' should be the only way to get an instance of it.

### Minimal reproducible example

n/a

### Versions

- Vaadin / Flow version: 24.3.3
- Java version: n/a
- OS version: n/a
- Browser version (if applicable): n/a
- Application Server (if applicable): n/a
- IDE (if applicable): n/a

**Comment by mshabarov** [comment]
Yes, the public CTOR can be deprecated and later removed (even in the next minor).
